src/iidm/VoltageLevel.cpp

```diff
@@ -262,6 +279,12 @@ VoltageLevel& VoltageLevel::setLowVoltageLimit(double lowVoltageLimit) {
     return *this;
 }
 
+void VoltageLevel::setNetworkRef(Network& network) {
+    if (static_cast<bool>(m_networkRef)) {
+        m_networkRef = network;
+    }
```

Comment:
To be discuss: I think the networkRef should be settable to null (empty reference) when the voltage is detached from the network

Reply:
The bug exists in java => I posted https://github.com/powsybl/powsybl-core/issues/1960 and https://github.com/powsybl/powsybl-iidm4cpp/issues/421 to fix it later